### PR TITLE
Added component to create an ad creative from new VAST model

### PR DESCRIPTION
**Pull request by balavor**
## Changes
- `AdCreative` is now a component .
- Added `width` and `height` in MP4 creative.
- `AdCreative's` associated value is now an array of creatives instead of a single one. There're some changes done in code but it is **temporary** to make it work and build.  

@VerizonAdPlatforms/mobile-sdk-developers: Please review.

### CODE REVIEW

File: PlayerCore/SelectedAdCreative.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+import Foundation
+
+public struct SelectedAdCreative {
+    let creative: AdCreative
```
Comment (VladyslavAnokhin) [inline PlayerCore/SelectedAdCreative.swift]: You have to clean this state with the same actions as `VRMFinalResult` does

File: PlayerCore/SelectedAdCreative.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                url: $0.url,
+                clickthrough: action.inlineVAST.clickthrough,
+                pixels: action.inlineVAST.pixels,
+                id: action.inlineVAST.id,
+                width: $0.width,
+                height: $0.height,
+                scalable: $0.scalable,
+                maintainAspectRatio: $0.maintainAspectRatio)
+        }
+        if mp4AdCreatives.isEmpty == false {
```
Comment (AndriiMoskvin) [inline PlayerCore/SelectedAdCreative.swift]: `guard`?

File: PlayerCore/SelectedAdCreative.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        let vpaidAdCreatives: [AdCreative.VPAID] = action.inlineVAST.vpaidMediaFiles.compactMap {
+            return .init(
+                url: $0.url,
+                adParameters: action.inlineVAST.adParameters,
+                clickthrough: action.inlineVAST.clickthrough,
+                pixels: action.inlineVAST.pixels,
+                id: action.inlineVAST.id)
+        }
+        if vpaidAdCreatives.isEmpty == false {
```
Comment (AndriiMoskvin) [inline PlayerCore/SelectedAdCreative.swift]: `guard`?

File: PlayerCoreTests/Components/SelectMediaFileComponentTestCase.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
+//  Copyright 2018, Oath Inc.
```
Comment (VladyslavAnokhin) [inline PlayerCoreTests/Components/SelectMediaFileComponentTestCase.swift]: 2019 :)

File: PlayerCoreTests/Components/SelectMediaFileComponentTestCase.swift [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+
+import XCTest
+import CoreMedia
```
Comment (VladyslavAnokhin) [inline PlayerCoreTests/Components/SelectMediaFileComponentTestCase.swift]: Why do you need `CoreMedia`?

File: PlayerCore/SelectedAdCreative.swift [hunk 6]
```diff
@@ -0,0 +1,42 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+import Foundation
+
+public struct SelectedAdCreative {
```
[the remark this answers is not in the document]
Comment (RomanTysiachnik) [reply PlayerCore/SelectedAdCreative.swift]: Good point

**Comment by RomanTysiachnik** [comment]
@AndriiMoskvin @VladyslavAnokhin RFR

File: PlayerCore/components/State.swift [hunk 7]
```diff
@@ -7,6 +7,7 @@ public struct State {
     public let ad: Ad
     public let adKill: AdKill
     public let adMaxShowTime: TimerSession
+    public let adCreative: AdCreative
```
Comment (AndriiMoskvin) [inline PlayerCore/components/State.swift]: `selectedAdCreative`?

**Pull request merged**